firrtl/src/test/scala/firrtlTests/SerializerSpec.scala

```diff
@@ -82,6 +82,50 @@ object SerializerSpec {
       "test"
     )
 
+  val probeModule: String =
+    """module probeModule :
+      |  input clock : Clock
+      |  input cond : UInt<1>
+      |  input in : UInt<8>
+      |  input inProbe : Probe<UInt<8>>
+      |  output out : UInt<8>
+
+      |  inst c of child
+      |  wire fooProbe : Probe<UInt<8>>
+      |  c.in <= in
+      |  define fooProbe = probe(c.in)
+      |  out <= read(out)
```

Comment:
Nits, if this being valid FIRRTL is important (documenting an example that's expected to work, even if a fragment)?
* `read` should take a Probe type of some kind (and probably not assign to itself).  Can be inline `read(probe(my_thing))`, FWIW.
* no wires of `Probe<T>`, consider an output port instead for the define target/destination/"LHS".

Reply:
Yes thank you for the "nits"! It's better for the examples to be as similar as possible to real/working use cases